backend/api/models.py

```
@@ -181,6 +181,30 @@ class Meta:
+        ('pending', 'Pending'),
+        ('completed', 'Completed'),
+        ('failed', 'Failed'),
+    ]
+    
+    video = models.ForeignKey(Video, on_delete=models.CASCADE, related_name='webcam_recordings')
+    recorder = models.ForeignKey(User, on_delete=models.CASCADE)
+    filename = models.CharField(max_length=255)
+    recording_date = models.DateTimeField(auto_now_add=True)
+    upload_status = models.CharField(max_length=20, choices=UPLOAD_STATUS_CHOICES, default='pending')
+    upload_completed_at = models.DateTimeField(null=True, blank=True)
+    recording_url = models.URLField(max_length=500, blank=True, null=True)
```

Ruff (0.8.2)

Avoid using `null=True` on string-based fields such as `URLField`

(DJ001)